sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -1317,8 +1341,10 @@ private String getDocumentMetadata(String document) throws IOException {
                   spec.getTypeFn() != null ? spec.getTypeFn().apply(parsedDocument) : null,
                   spec.getIdFn() != null ? spec.getIdFn().apply(parsedDocument) : null,
                   spec.getUsePartialUpdate() ? DEFAULT_RETRY_ON_CONFLICT : null);
+          SimpleModule module = new SimpleModule();
+          module.addSerializer(DocumentMetadata.class, new DocumentMetadataSerializer());
+          OBJECT_MAPPER.registerModule(module);
```

Comment:
This shouldn't be done **here**, should it?

Reply:
Correct, thanks for pointing out, it should not be done for every element processed

Reply:
@echauchot: from playing with it I can say that this is a critical issue for a production pipeline because of fighting for memory that starts to occur very soon after starting a streaming job (depending on the workload, etc., of course) and results in a sudden dramatic slowdown of the pipeline.

Reply:
yeah, sure, can you take a look at the fix https://github.com/apache/beam/pull/11504 and review it ?

Reply:
I can see that it is merged already. LGTM too. Thanks!